src/stats.rs

```diff
@@ -150,6 +150,47 @@ pub struct MediaIngressStats {
     // pub remote: RemoteEgressStats,
 }
 
+impl MediaIngressStats {
+    /// Merge `other` into `self`, mutating `self`.
+    ///
+    /// **Panics** if called with stats that don't have the same `(mid, rid)` pair.
+    pub(crate) fn merge(&mut self, other: &Self) {
+        *self = self.merged(other);
+    }
+
+    /// Merge `self` and `other` returning a new `Self` of the result.
+    ///
+    /// **Panics** if called with stats that don't have the same `(mid, rid)` pair.
+    fn merged(&self, other: &Self) -> Self {
+        assert!(
+            self.mid == other.mid,
+            "Cannot merge MediaIngressStats for different mids"
+        );
+        assert!(
+            self.rid == other.rid,
+            "Cannot merge MediaIngressStats for different rids"
+        );
+        let (rtt, loss) = if self.timestamp > other.timestamp {
+            (self.rtt, self.loss)
+        } else {
+            (other.rtt, other.loss)
+        };
+
+        Self {
+            mid: self.mid,
+            rid: self.rid,
+            bytes: self.bytes + other.bytes,
+            packets: self.packets + other.packets,
+            firs: self.firs + other.firs,
+            plis: self.plis + other.plis,
+            nacks: self.nacks + other.nacks,
+            rtt,
+            loss,
+            timestamp: self.timestamp.max(other.timestamp),
+        }
+    }
```

Comment:
Why is this split into two functions? Seems like we either modify-in-place or we don't. But here we modify in place by using a function that doesn't. Doesn't it also create an extra allocation that we just throw away?

Reply:
Another way to not forget fields is to deconstruct `other`

```rust
let Self  { _all the fields_ } = other;
```

Reply:
Yeah I suppose, it occupies some local names that I'd want though.

I got rid of `merged` in the latest commit but kept the `*self = Self { ..}` bit